### PR TITLE
Optimizing required pods logging

### PR DESCRIPTION
**Pull request by jinronga**
**Notes for Reviewers**

This is an optimized code submission.User retrieves specified pod log information and checks whether the required pod exists.The original judgment was a character array traversal comparison. After optimization, the number of for loops was reduced to save time complexity.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit f6bc02de29614e91cc27744b41f94b9280b367bc at: https://649718dbdd2097762bda655a--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1bd3ea07e44e47c6b8d231b02b0fd67c289513cd at: https://6497c0ce5380a24937cb9745--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1bd3ea07e44e47c6b8d231b02b0fd67c289513cd at: https://6497c0dcf3399c4ade34ba90--meshery-docs-preview.netlify.app

File: docs/pages/reference/mesheryctl/test.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# test
+
+
+
+## Options
+
+<pre class='codeblock-pre'>
+<div class='codeblock'>
+  -h, --help   help for test
```
Comment (alphaX86) [inline docs/pages/reference/mesheryctl/test.md:21]: @jinronga was this page auto-generated during the test runs? 👀

File: ui/package-lock.json
Comment (alphaX86) [inline ui/package-lock.json:1]: @Abhishek-kumar09 ?

File: docs/pages/reference/mesheryctl/test.md (hunk 1 above)
Comment (jinronga) [reply docs/pages/reference/mesheryctl/test.md:21]: Yes, it seems so. Sorry, I didn't pay attention.

**Comment by l5io** [comment]
🚀 Preview for commit c0f0e97c90e098f7d847d3393b227506dc0150dd at: https://6498e7f993a1cb63d6b147f6--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 86c73f424fb266a17982c5510c96d9ff8f10ad09 at: https://649a9f4dbc5da15ca2e671ab--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
@jinronga Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by jinronga** [comment]
Sorry,&nbsp;I&nbsp;am&nbsp;not&nbsp;available&nbsp;for&nbsp;a&nbsp;meeting&nbsp;at&nbsp;the&nbsp;moment.&nbsp;Can&nbsp;we&nbsp;communicate&nbsp;on&nbsp;Slack&nbsp;instead?




笙
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "meshery/meshery"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月28日(星期三) 下午4:41
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [meshery/meshery] Optimizing required pods logging (PR #7932)





 
@jinronga Let's discuss this on Meshery Dev call. Please add this as an agenda item in the meeting minutes if you would. :)

**Comment by l5io** [comment]
🚀 Preview for commit 0571606065e0b6e03ca563ac4fc1ee48944e5469 at: https://649c136a6599ef0e9709c8aa--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 0571606065e0b6e03ca563ac4fc1ee48944e5469 at: https://649c136dc242c80e831d5564--meshery-docs-preview.netlify.app

File: docs/pages/reference/mesheryctl/test.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/reference/mesheryctl/test.md:21]: @jinronga please remove this file / page.

File: mesheryctl/doc/cmds.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+permalink: reference/cmds.yml/main
+redirect_from: reference/cmds.yml/main/
+type: reference
+display-title: "false"
+language: en
+command: cmds.yml
+subcommand: nil
+---
+
+name: test
```
Comment (leecalcote) [inline mesheryctl/doc/cmds.yml:13]: @alphaX86 are each of these "test" section names appropriate?

File: mesheryctl/pkg/utils/platform.go [hunk 3]
```diff
@@ -653,6 +653,24 @@ func GetRequiredPods(specifiedPods []string, availablePods []v1core.Pod) ([]stri
...
+	requiredPodsMap := make(map[string]string)
+	var availablePodsName []string
+	for _, pod := range availablePods {
+		availablePodsName = append(availablePodsName, pod.GetName())
+	}
+	for _, sp := range specifiedPods {
+		if index := StringContainedInSlice(sp, availablePodsName); index != -1 {
+			requiredPodsMap[availablePodsName[index]] = availablePodsName[index]
+		} else {
+			return nil, fmt.Errorf("pod \"%s\" specified, does not exist in the 'meshery' namespace. Please  Run `mesheryctl system status` to view the available pods", sp)
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/platform.go:667]: @alphaX86 have we hardcoded support only deploying Meshery to a namespace titled `meshery`?

File: ui/package-lock.json
Comment (leecalcote) [inline ui/package-lock.json:1]: @jinronga do you mind removing this file from the PR?

**Comment by leecalcote** [comment]
Thanks for this PR, @jinronga! 👍

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: mesheryctl/pkg/utils/platform.go (hunk 3 above)
Comment (jinronga) [reply mesheryctl/pkg/utils/platform.go:667]: Ok. I have removed.

File: ui/package-lock.json
Comment (jinronga) [reply ui/package-lock.json:1]: I don't mind.

Comment (Abhishek-kumar09) [reply ui/package-lock.json:1]: Its Ok 👍

**Comment by Chadha93** [comment]
@jinronga Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by alphaX86** [comment]
Repeated PR. Already fixed via #8067. Closing this...

**Pull request closed without merging**